### PR TITLE
Add Reproducibility Instructions and Unzip Data

### PR DESCRIPTION
**Pull request by gwaygenomics**
The following pull request adds, fixes, and/or modifies:

* [M] `1.download-data/downloadData.py` - adding unzip argument and updating download path
* [M] `README` - adding instructions on how to reproduce compute environment and execute analysis pipeline
* [M] `1.download-data/README` - updating download execution instructions
* [A] `1.download-data/scripts/unzip_data.py` - adding function to unzip downloaded data
* [A] `environment.yml` - the initial conda environment
* [A] `analysis.sh` - the initial bash script storing analysis reproducibility instructions
* [A] `1.download-data/download-pipeline.sh` - bash script that, when executed, will download and extract all data
* [A] `1.download-data/md5sums.txt` - file storing md5sums for all files for download and extraction integrity checks

I performed the following prior to filing this pull request:

- [x] Tested that my change does not break the analysis pipeline
- [x] Ran a linter through my code
- [x] Update environment dependencies if my code introduces a new package (**Added environment here!**)

### CODE REVIEW

**Comment by gwaygenomics** [comment]
cc @wagenrace - can review these changes?

**Comment by wagenrace** [comment]
@gwaygenomics what is best to do with windows users? Write instruction how to use WSL ubuntu or look for a different solution as .sh?

File: 1.download-data/scripts/unzip_data.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    # Ensure that the file exists and is fully downloaded
+    if os.path.exists(locationData):
+
+        fileComplete = checkIfDownloadComplete(locationData)
+        assert fileComplete, "Found file seems to be incomplete, please download again"
+
+        with ZipFile(locationData, "r") as zip_fh:
+            zip_fh.printdir()
+            print("Extracting all files")
+            zip_fh.extractall(path="data")
```
[the remark this answers is not in the document]
Comment (gwaygenomics) [reply 1.download-data/scripts/unzip_data.py]: yeah, I agree - I will add!

**Comment by gwaygenomics** [comment]
Thanks for the review @wagenrace - in a6ada432e446fca018f2d0b309b043151795342b i remove the zip file after file extraction. If everything else looks ok, I will go ahead and merge!

**Comment by wagenrace** [comment]
With the idea of 'self contained modules' I do understand the choices made. I will keep them in mind for the further modules

About windows, my own machine is a windows machine where I installed Ubuntu on. I will write an instruction to run it this way and see it outside the scope of this pull request.

I have ran it one more time with the change on my computer to make sure everything works fine and we don't have a 'it worked on my machine' bug. Everything works fine!

@gwaygenomics go ahead and merge

**Comment by gwaygenomics** [comment]
> About windows, my own machine is a windows machine where I installed Ubuntu on. I will write an instruction to run it this way and see it outside the scope of this pull request.

I love the idea of being OS agnostic and any instructions on how to get this up and running on a windows machine would be great. Feel free to create an issue for this feature request (and then close the issue once you've implemented it)

> I have ran it one more time with the change on my computer to make sure everything works fine and we don't have a 'it worked on my machine' bug. Everything works fine!

🎉 I will merge

**Comment by gwaygenomics** [comment]
> Feel free to create an issue for this feature request (and then close the issue once you've implemented it)

Nevermind! I see it is posted in #11 :)

**Pull request merged**